### PR TITLE
add a job config for garbage collector

### PR DESCRIPTION
**Pull request by caesarxuchao**
I plan to test the garbage collector using the kubemark framework. Most fields are copied from the kubermark configuration.

@gmarek @ixdy

### CODE REVIEW

**Comment by ixdy** [comment]
Sorry for the delay. This basically looks fine, though we've refactored a few things since you opened this PR (I think mostly relating to `provider-env`?), so it's currently broken.

If you copy-paste the current job template from kubemark, it might just work.

Alternately, maybe just add the jobs only to the `kubernetes-kubemark.yaml` file, using a suffix like `garbagecollector-5-gce` to differentiate vs. the normal kubemark jobs?

**Comment by caesarxuchao** [comment]
I updated the template. Test passed. PTAL. Thank you!

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-garbagecollector.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+- project:
+    name: kubernetes-garbage-collector
+    suffix:
+        - '5-gce':
+            description: 'Run minimal Kubemark to make sure it is not broken.'
+            timeout: 60
+            cron-string: '{sq-cron-string}'
+            job-env: |
+                export E2E_NAME="kubemark-5"
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-garbagecollector.yaml]: nit: maybe use something like garbagecollector-5 ?

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-garbagecollector.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        - workspace-cleanup:
+            dirmatch: true
+            external-deletion-command: 'sudo rm -rf %s'
+        - e2e-credentials-binding
+
+- project:
+    name: kubernetes-garbage-collector
+    suffix:
+        - '5-gce':
+            description: 'Run minimal Kubemark to make sure it is not broken.'
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-garbagecollector.yaml]: this job looks exactly the same as the regular kubemark-5 job. do we need it?

Comment (caesarxuchao) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-garbagecollector.yaml]: Removed.

**Comment by caesarxuchao** [comment]
Yes, the actual tests are not merged yet.

**Comment by fejta** [comment]
FYI @spxtr I'm merging this.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-garbagecollector.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        - timestamps
+        - workspace-cleanup:
+            dirmatch: true
+            external-deletion-command: 'sudo rm -rf %s'
+        - e2e-credentials-binding
+
+- project:
+    name: kubernetes-garbage-collector
+    suffix:
+        - '100-gce':
```
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-garbagecollector.yaml]: Please add a comment with the full name of this job (like others do)

**Comment by fejta** [comment]
@caesarxuchao ping me after you add the comment to the job name, thanks!

**Comment by caesarxuchao** [comment]
@fejta Done. Thanks!

**Pull request merged**